### DISCUSSION TITLE
There has a way to decode encoded types from another language?

### DISCUSSION THREAD

**Discussion by Keony1** [kriszyp/msgpackr#99] [Q&A]
I was trying to decode a type that was encoded in C# and compressed with lz4 and saved at redis, without success as you can imagine.
So I get to the point to wonder if it is really possible, because the type number and how the decoding works.

I get the register like this (a small example in utf8)

```json
[
    {
        "type": 98,
        "data": {
            "0": 205,
            "1": 12,
            "2": 179
        }
    },
    {
        "0": 243,
        "1": 32,
        "2": 141,
        "3": 174,
        "4": 71,
        // ....
       // and so on
  }
]
````

first tried to parse the entire JSON into Buffer and then decode... without success of course.

second tried to parse only the first object of the array, because it have the type number. 
I supposed I would be able to decode the data object, but you know... without success.

third tried to parse only the second object of the array, and yep... without success

So it is possible to decode types from another language? I need to mimic the type and register with `addExtension`?
but if I have to do this,  how  `decode` will identify the type number?

**Comment by kriszyp** [comment]
MessagePack is language agnostic, so yes, you should be able to encode in one language and decode in another. But I don't know what I am looking at here. 98 in MessagePack is just the unsigned integer 98 (with no further data), so I doubt this is actually MessagePack. Do you need to decompress the data with lz4 first?

**Reply by Keony1** [reply]
Thank you for your time to answer me.
I realized what is going on. They're using a MessagePack library in C# and passing a parameter called `TypelessContractlessStandardResolver` and, as the lib documentation words: `Consider the result to be a .NET-specific MessagePack binary [...]`

I suppose we can assume `98` is the type number of this specific feature. That's the reason I couldn't decode the binary.
